crypto/schemes.go

```diff
@@ -0,0 +1,138 @@
+	SigGroup kyber.Group
+	// KeyGroup is the group used to create the keys
+	KeyGroup kyber.Group
+	// ThresholdScheme is the signature scheme used, defining over which curve the signature
+	// and keys respectively are.
+	ThresholdScheme sign.ThresholdScheme
+	// AuthScheme is the signature scheme used to identify public identities
+	AuthScheme sign.AggregatableScheme
+	// DKGAuthScheme is the signature scheme used to authenticate packets during broadcast in a DKG
+	DKGAuthScheme sign.Scheme
+	// Pairing is the underlying pairing Suite.
+	Pairing pairing.Suite
+	// the hash function used by this scheme
+	HashFunc func() hash.Hash
+}
+
+type schnorrSuite struct {
+	kyber.Group
+}
+
+func (s *schnorrSuite) RandomStream() cipher.Stream {
+	return random.New()
+}
+
+//nolint:dupl
+func NewPedersenBLSChained() (cs *Scheme) {
+	var Pairing = bls.NewBLS12381Suite()
+	var KeyGroup = Pairing.G1()
+	var SigGroup = Pairing.G2()
+	var ThresholdScheme = tbls.NewThresholdSchemeOnG2(Pairing)
+	var AuthScheme = signBls.NewSchemeOnG2(Pairing)
+	var DKGAuthScheme = schnorr.NewScheme(&schnorrSuite{KeyGroup})
+	var HashFunc = func() hash.Hash { h, _ := blake2b.New256(nil); return h }
+
+	return &Scheme{
+		Name:                     "pedersen-bls-chained",
+		IsPreviousSigSignificant: true,
+		SigGroup:                 SigGroup,
+		KeyGroup:                 KeyGroup,
+		ThresholdScheme:          ThresholdScheme,
+		AuthScheme:               AuthScheme,
+		DKGAuthScheme:            DKGAuthScheme,
+		Pairing:                  Pairing,
+		HashFunc:                 HashFunc,
+	}
+}
+
+//nolint:dupl
+func NewPedersenBLSUnchained() (cs *Scheme) {
+	var Pairing = bls.NewBLS12381Suite()
+	var KeyGroup = Pairing.G1()
+	var SigGroup = Pairing.G2()
+	var ThresholdScheme = tbls.NewThresholdSchemeOnG2(Pairing)
+	var AuthScheme = signBls.NewSchemeOnG2(Pairing)
+	var DKGAuthScheme = schnorr.NewScheme(&schnorrSuite{KeyGroup})
+	var HashFunc = func() hash.Hash { h, _ := blake2b.New256(nil); return h }
+
+	return &Scheme{
+		Name:                     "pedersen-bls-unchained",
+		IsPreviousSigSignificant: false,
+		SigGroup:                 SigGroup,
+		KeyGroup:                 KeyGroup,
+		ThresholdScheme:          ThresholdScheme,
+		AuthScheme:               AuthScheme,
+		DKGAuthScheme:            DKGAuthScheme,
+		Pairing:                  Pairing,
+		HashFunc:                 HashFunc,
+	}
+}
+
+//nolint:dupl
+func NewPedersenBLSUnchainedSwapped() (cs *Scheme) {
+	var Pairing = bls.NewBLS12381Suite()
+	var KeyGroup = Pairing.G2()
+	var SigGroup = Pairing.G1()
+	var ThresholdScheme = tbls.NewThresholdSchemeOnG1(Pairing)
+	var AuthScheme = signBls.NewSchemeOnG1(Pairing)
+	var DKGAuthScheme = schnorr.NewScheme(&schnorrSuite{KeyGroup})
+	var HashFunc = func() hash.Hash { h, _ := blake2b.New256(nil); return h }
+
+	return &Scheme{
+		Name:                     "pedersen-bls-unchained-swapped",
```

Comment:
This should be added to `common/scheme/scheme.go`.

Reply:
`common/scheme/scheme.go` is now gone

Reply:
I can see it being used and it's changed in this PR too.
Maybe you haven't pushed the change yet?